Binaries/DafnyRuntime.h

```diff
@@ -324,18 +349,18 @@ struct DafnySequence {
       start[i] = t;
     }
 
-    explicit DafnySequence(shared_ptr<vector<T>> arr) {
-      len = arr->size();
+    explicit DafnySequence(DafnyArray<T> arr) {
+      len = arr.size();
       sptr = shared_ptr<T> (new T[len], std::default_delete<T[]>());
       start = &*sptr;
-      std::copy(arr->begin(), arr->end(), start);
+      std::copy(arr.vec->begin(), arr.vec->end(), start);
```

Comment:
Should we add `begin` and `end` functions to the DafnyArray struct, so clients don't have to care if we decide to change the representation again?